pandas/core/internals.py

```diff
@@ -3489,7 +3484,8 @@ def apply(self, f, axes=None, filter=None, do_integrity_check=False,
         # with a .values attribute.
         aligned_args = dict((k, kwargs[k])
                             for k in align_keys
-                            if hasattr(kwargs[k], 'values'))
+                            if hasattr(kwargs[k], 'values') and
+                            not isinstance(kwargs[k], ABCExtensionArray))
```

Comment:
If an ExtensionArray chooses to store it's data as `.values`, `setitem` would be broken without this extra check.

Reply:
Should we make a test for this? (eg call the underlying data `.values` in of the example test arrays?

Reply:
`_values`

Reply:
this is pretty special casey here. shouldn't this check for ``._values``?

Reply:
I'm not sure. I don't really know could be in kwargs. You think it's only ever Index or Series? Or could it be a dataframe or block?